src/services/MnemonicSave.ts

```diff
@@ -0,0 +1,66 @@
+import { Storage } from "@utils/StorageArray";
+
+const CHECK_NEEDED_KEY = "mnemonic-check-needed";
+
+interface WordChecked {
+  word: string;
+  isDisabled: boolean;
+}
+export class MnemonicSave {
+  public mnemonicArr: string[];
+  public sortedMnemonic: WordChecked[];
+  private counter: number;
+  constructor(private readonly storage: Storage) {
+    this.mnemonicArr = [];
+    this.sortedMnemonic = [];
+    this.counter = 0;
```

Comment:
I still really don't like maintaining the state in a global service.

I don't have a problem with a supporting class defined specifically for the state for the component. That can be unit tested and isolated. But importantly, there would be one instance of the class for each component instance. My problem with the state here is that it is global, which is unnecessary complexity.